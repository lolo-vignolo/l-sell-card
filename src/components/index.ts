export { ProductButtons } from '../components/ProductButtons';
export { ProductImage } from '../components/ProductImage';
export { ProductTitle } from '../components/ProductTitle';
export { ProductCard } from '../components/ProductCard';
