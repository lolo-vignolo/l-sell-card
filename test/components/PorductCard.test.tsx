import React from 'react';
import render from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { product_1 } from '../data/product';

const { act } = render;

describe('PoductTitle', () => {
  test('children se renderiza correctamente', () => {
    const wrapper = render.create(
      <ProductCard product={product_1}>
        {() => (
          <>
            <h1>gola mundo</h1>
          </>
        )}
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe incremenat el contador', () => {
    const wrapper = render.create(
      <ProductCard product={product_1}>
        {({ handleIncrease, count }) => (
          <>
            <h1>Product card</h1>
            <span>{count}</span>
            <button onClick={() => handleIncrease(1)}>incrementar</button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    });
    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('1');
  });
});
