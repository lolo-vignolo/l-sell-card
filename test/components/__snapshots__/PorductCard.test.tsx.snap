// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoductTitle children se renderiza correctamente 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    gola mundo
  </h1>
</div>
`;

exports[`PoductTitle debe incremenat el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    incrementar
  </button>
</div>
`;

exports[`PoductTitle show the component rendered correctly 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    gola mundo
  </h1>
</div>
`;
